compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64MacroAssembler.java

```diff
@@ -1802,28 +1813,24 @@ public void fmadd(int size, Register dst, Register src1, Register src2, Register
      * Compares x and y and sets condition flags.
      *
      * @param size register size. Has to be 32 or 64.
-     * @param x general purpose register. May not be null or stackpointer.
+     * @param x general purpose register. May not be null.
      * @param y general purpose register. May not be null or stackpointer.
      */
     public void cmp(int size, Register x, Register y) {
         assert size == 32 || size == 64;
-        super.subs(size, zr, x, y, ShiftType.LSL, 0);
+        subs(size, zr, x, y);
```

Comment:
Why drop the super. here? Is it because x might be sp?

Reply:
Yes, I dropped the super so that it can call the aarch64macroassembler subs, which handles more combinations of xzr and sp